revive.toml

```diff
@@ -0,0 +1,104 @@
+[rule.context-as-argument]
+[rule.context-keys-type]
+[rule.dot-imports]
+[rule.error-naming]
+[rule.error-return]
+[rule.error-strings]
+[rule.errorf]
+# [rule.exported] # disabled due to lack of value / encouraging bad habits.  it complains about internal packages.
+[rule.if-return]
+[rule.increment-decrement]
+[rule.indent-error-flow]
+# [rule.package-comments] # disabled due to lack of value / encouraging bad habits.  it complains about internal packages.
+[rule.range]
+[rule.receiver-naming]
+[rule.time-naming]
+[rule.unexported-return]
+[rule.var-declaration]
+[rule.var-naming] # noisy, but would be good to fix these in v2
+
+#### higher value stuff
+
+# this is basically errcheck, warns on errs that are not checked.
+# strongly desired, but disabled due to 300 failures (to be tackled incrementally).
+# [rule.unhandled-error]
+
+# general defer gotchas.
+#
+# in particular: "recover" warns about unsafe use of recover().
+# this has caught bugs that can allow crashes while seemingly safe, and are *extremely* hard to catch in review.
+#
+# the arguments are excluding only "call-chain", which would disallow `defer someFn(...)()` which is both useful and in use.
+[rule.defer]
+arguments=[["loop","method-call","recover","return", "immediate-recover"]]
+
+# string(int) is almost always a bug.
+# go vet considers this a fatal error, but only in 1.15 or newer, and go.mod currently targets 1.13
+[rule.string-of-int]
+
+#### added because we currently have zero violations, and they seem decent enough to retain
+
+[rule.atomic] # correct use of sync code, important
+[rule.call-to-gc] # beneficial
+[rule.constant-logical-expr] # minor code simplifier
+[rule.identical-branches] # code simplifier / failures are pretty dubious
+[rule.modifies-parameter] # beneficial
+[rule.modifies-value-receiver] # probably beneficial, prevents subtle bugs
+[rule.range-val-address] # beneficial
+[rule.range-val-in-closure] # beneficial
+[rule.unconditional-recursion] # probably a good idea
+[rule.unreachable-code] # code simplifier
+[rule.waitgroup-by-value] # correct use of sync code, important
+
+#### unused utilities
+
+# [rule.file-header] # could possibly replace `copyright -verifyOnly`?
+# [rule.imports-blacklist] # simple way to ban imports / enforce wrappers, likely useful
+
+#### disabled but maybe desirable
+
+# [rule.bare-return] # probably beneficial as it's slightly error-prone, but 2,000 failures
+# [rule.bool-literal-in-expr] # minor code simplifier, few failures
+# [rule.confusing-results] # maybe beneficial, only a few failures
+# [rule.deep-exit] # probably a good idea in most code, some failures, but not trivial to adopt
+# [rule.duplicated-imports] # minor, but may be worthwhile.  failures are weird but harmless
+# [rule.early-return] # minor code simplifier, a handful of failures
+# [rule.get-return] # existing failures are intentional + desirable, but in principle it's a fine idea
+# [rule.import-shadowing] # probably beneficial, but 750 failures
```

Comment:
:/ 

the shadowing rules continue to cause me pain over and over again. Hopefully one-day we can enable

Reply:
definitely agreed.  it makes cmd-f a lot less useful too.